adapters/tpmn/tpmn.go

```diff
@@ -0,0 +1,143 @@
+package tpmn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// TpmnAdapter struct
+type adapter struct {
+	uri string
+}
+
+// MakeRequests makes the HTTP requests which should be made to fetch bids from TpmnBidder.
+func (rcv *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	validImps, errs := getValidImpressions(request, reqInfo)
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	request.Imp = validImps
+
+	requestBodyJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  http.MethodPost,
+		Uri:     rcv.uri,
+		Body:    requestBodyJSON,
+		Headers: headers,
+	}}, errs
+}
+
+// getValidImpressions validate imps and check for bid floor currency. Convert to EUR if necessary
+func getValidImpressions(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]openrtb2.Imp, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+
+	for _, imp := range request.Imp {
+		if err := preprocessBidFloorCurrency(&imp, reqInfo); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := preprocessExtensions(&imp); err != nil {
+			errs = append(errs, err)
+			continue
+		}
```

Comment:
While taking a final look, this check seems unnecessary. Usually, you would like to process the extension if you want to use the fields from `adapters.ExtImpBidder` in your adapter code and that is clearly not the case here. The code is just doing json unmarshal. I would recommend to remove this check. thanks!

Reply:
I agree with what you said. After receiving and checking the message, I found out that the prebid server already does the check for the presence of the common extension filed by default. it is code unnecessary. I have removed the code as requested. Additionally, the 'no-imp-ext.json' test file has also been modified.

update commit : [3f5af95](https://github.com/prebid/prebid-server/pull/2947/commits/3f5af95e6f53af9f0df0f8aaa84b2cd94e644362)